kadi/occweb.py

```diff
@@ -11,29 +11,38 @@
 import time
 from collections import OrderedDict as odict
 from pathlib import Path
+import logging
 
 import numpy as np
 import requests
 
 from Chandra.Time import DateTime
 from astropy.io import ascii
+from astropy.table import Table
+from astropy.utils.data import download_file
 
+# This is for deprecated functionality to cache to a local directory
 CACHE_DIR = 'cache'
 CACHE_TIME = 86000
 TIMEOUT = 60
 
-ROOTURL = 'http://occweb.cfa.harvard.edu'
+ROOTURL = 'https://occweb.cfa.harvard.edu'
 URLS = {'fdb_major_events': '/occweb/web/fdb_web/Major_Events.html',
         'fot_major_events': '/occweb/web/fot_web/eng/reports/Chandra_major_events.htm',
         'ifot': '/occweb/web/webapps/ifot/ifot.php',
         }
 
+# Initialize 'kadi.occweb' logger.
+logger = logging.getLogger(__name__)
+
+
+def get_auth(username=None, password=None):
+    if username and password:
+        return (username, password)
```

Comment:
Not entirely sure I understand this use case.

Reply:
This allows the user / calling function to explicitly pass in credentials instead of getting them automatically through other means. See e.g. `get_occweb_page`.

Reply:
Right, it just seems like `get_occweb_page` could have the if statement and then not hand the username and password to a routine just to get them back.

Reply:
But this function handles a middle-ground case of `username` being `None`. So the API really needs to provide the user, passwd and return a user, passwd pair. 

It's really just simplest to have a single function that does *everything* instead of delegating some of that user/pass logic outside. And if we ever wrote another function that needs OCCweb auth handling then that function would need to repeat the special handling of both user and passwd being provided.